### ISSUE TITLE
Restructure Quo2 component library

### ISSUE THREAD

**Issue by J-Son89** [status-im/status-react#13903]
# Problem

Currently there is not much structure within the quo2 component library. It is a flat list of components.

## Implementation

There has been consensus among the dev and design team to have a 1-1 mapping of the components used to the Figma design categorisation. (Unless it makes sense for the component to live elsewhere)

## Acceptance Criteria

- Restructure the current component library in code to match the Figma mapping
 e.g 
 ```
 src/quo2/components
   |--/tags
        |-- tag
        |-- tags
        |-- context-tags
   |--/avatar
        |-- ring
        |-- user avatar
        |-- group avatar
   |--/switcher
        |-- home-cards
        |-- 1-1-message-card
   ```
- Restructure the quo2 component library in the app to match this structure too.
- Add some documentation/ design decisions in the relevant location.

## Future Steps

Ensure future p.r's have components being added in a similar manner.

**Comment by chrispader** [comment]
Also naming of some components might be inconsistent. E.g. is `icon_avatar` the same as `channel_avatar`? Or are there more use cases for `icon_avatar` than for channels?

**Comment by chrispader** [comment]
Edit: Should we use `icon_avatar` component for `channel_avatar`?

**Comment by J-Son89** [comment]
> Edit: Should we use `icon_avatar` component for `channel_avatar`?

Thanks Chris, I will bring this conversation up as I'm doing the work

**Comment by J-Son89** [comment]
> Edit: Should we use `icon_avatar` component for `channel_avatar`?

I checked this component and Icon component is accurately named. 
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=102%3A5246

However I did notice that channel avatar is extremely similar and functionally these could be done by the same component. Perhaps more developer input could be contributed to the design of the components so we don't have to make multiple implementations of essentially the same component. The tags are a similar case.
